packages/react-scripts/scripts/build.js

```diff
@@ -142,6 +142,11 @@ function build(previousSizeMap) {
       process.exit(1);
     }
 
+    if (process.env.CI && stats.compilation.warnings.length) {
+     printErrors('Failed to compile. Note, the build has crashed because it is being run with the environment variable CI set to true. In this mode the build crashes when any warnings are encountered.', stats.compilation.warnings);
```

Comment:
> Note, the build has crashed because it is being run with the environment variable CI set to true.

I think this wording makes it sound like having the `CI` env variable set to true is the problem, not the warnings themself. I hope people won't "fix" this by trying to set the env variable to false...